zerver/tests/test_messages.py

```diff
@@ -1066,6 +1067,32 @@ def test_message_mentions(self) -> None:
         message = most_recent_message(user_profile)
         assert(UserMessage.objects.get(user_profile=user_profile, message=message).flags.mentioned.is_set)
 
+    @mock.patch("zerver.lib.actions.send_event")
+    def test_user_group_mentions(self, mock_send_event: mock.MagicMock) -> None:
+        iago = self.example_user('iago')
+        hamlet = self.example_user('hamlet')
+        zoe = self.example_user("ZOE")
+
+        user_group_a = create_user_group('user_group_a', [iago], get_realm('zulip'))
+        user_group_b = create_user_group('user_group_b', [iago, hamlet, zoe], get_realm('zulip'))
+        self.subscribe(iago, 'Denmark')
+        self.subscribe(hamlet, 'Denmark')
+
+        self.send_stream_message(self.example_user('cordelia'), 'Denmark',
+                                 content='Hi, @*user_group_a* @*user_group_b* @**Zoe**')
+
+        # Extract the send_event call and look for the `mentioned_group`
+        # property in the notified user arguments.
+        for call_args in mock_send_event.call_args_list:
+            (_, _, notified_users_arg) = call_args[0]
+            for user_arg in notified_users_arg:
+                if user_arg['id'] == zoe.id:
+                    self.assertEqual(user_arg['mentioned_group'], None)
+                if user_arg['id'] == iago.id:
+                    self.assertEqual(user_arg['mentioned_group'], user_group_a.id)
+                if user_arg['id'] == hamlet.id:
+                    self.assertEqual(user_arg['mentioned_group'], user_group_b.id)
+
```

Comment:
@Hypro999 I have updated the test function. Kindly have a look.

Reply:
This test looks great! I have a _small_ suggestion though. Just for the sake of completeness (so that when someone else sees this test they know exactly how to extract the data they want), when you do the the `self.assertEqual(user_arg['mentioned_group'] ...` check, can you also follow up with checking the `flags` to see that `mentioned` is included. This way someone reading the test can see that, for example, `zoe` was mentioned because `flags['mentioned'] == True` but not because of a group (because `user_arg['mentioned_group'] == None`).

Reply:
@Hypro999 done. Thanks!